### PR TITLE
feat: sensor GUI

### PR DESCRIPTION
**Pull request by whynowy**
GUI for Sensor related. Dependencies editor is not there, as it's a little bit complicated which needs to consider the `filters` of a dependency, it could be an enhancement later.

Signed-off-by: Derek Wang <whynowy@gmail.com>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).\

### CODE REVIEW

**Comment by whynowy** [comment]
g01
g02
g03

File: server/sensor/sensor_server_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package sensor
```
Comment (alexec) [inline server/sensor/sensor_server_test.go:1]: delete this empty fille

File: ui/src/app/sensors/components/sensor-details/sensor-details.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                                    .then(() => setEdited(false))
+                                    .then(() => setError(null))
+                                    .catch(setError)
+                        },
+                        {
+                            title: 'Delete',
+                            iconClassName: 'fa fa-trash',
+                            disabled: edited,
+                            action: () => {
+                                if (!confirm('Are you sure you want to delete this sensor object?\nThere is no undo.')) {
```
Comment (alexec) [inline ui/src/app/sensors/components/sensor-details/sensor-details.tsx:81]: minor - `popup.confirm` is a more aesthetically pleasing dialag

File: ui/src/app/sensors/components/sensor-list/sensor-list.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@import 'node_modules/argo-ui/src/styles/config';
+
+.argo-table-list {
```
Comment (alexec) [inline ui/src/app/sensors/components/sensor-list/sensor-list.scss:3]: how come we need this?

File: ui/src/app/sensors/components/sensor-list/sensor-list.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                        <Tabs
+                            navTransparent={true}
+                            selectedTabKey={tab}
+                            onTabSelected={setTab}
+                            tabs={[
+                                {
+                                    title: 'LOGS',
+                                    key: 'logs',
+                                    content: (
+                                        <div>
```
Comment (alexec) [inline ui/src/app/sensors/components/sensor-list/sensor-list.tsx:182]: I think this should be a separate component

**Review by alexec — COMMENTED** [review]
only one big change requested - can we have the log viewer as a separate component?

File: ui/src/app/sensors/components/sensor-details/sensor-details.tsx (hunk 2 above)
Comment (whynowy) [reply ui/src/app/sensors/components/sensor-details/sensor-details.tsx]: done.

File: ui/src/app/sensors/components/sensor-list/sensor-list.scss (hunk 3 above)
Comment (whynowy) [reply ui/src/app/sensors/components/sensor-list/sensor-list.scss]: deleted.

File: ui/src/app/sensors/components/sensor-list/sensor-list.tsx (hunk 4 above)
Comment (whynowy) [reply ui/src/app/sensors/components/sensor-list/sensor-list.tsx]: 👍

File: ui/src/app/sensors/components/sensor-logs-viewer.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    selectedTrigger,
+    sensor,
+    onClick
+}: {
+    namespace: string;
+    selectedTrigger: string;
+    sensor: Sensor;
+    onClick: (selectedNdoe: string) => void;
+}) => {
+    const [error, setError] = useState<Error>();
```
Comment (alexec) [inline ui/src/app/sensors/components/sensor-logs-viewer.tsx:23]:
```
ERROR in /Users/acollins8/go/src/github.com/argoproj/argo/ui/src/app/sensors/components/sensor-logs-viewer.tsx
./src/app/sensors/components/sensor-logs-viewer.tsx
[tsl] ERROR in /Users/acollins8/go/src/github.com/argoproj/argo/ui/src/app/sensors/components/sensor-logs-viewer.tsx(23,12)
      TS6133: 'error' is declared but its value is never read.

```

Comment (alexec) [reply ui/src/app/sensors/components/sensor-logs-viewer.tsx:23]: you need an <ErrorNotice/> somewhere I think

File: server/sensor/sensor_server.go [hunk 6]
```diff
@@ -28,6 +27,29 @@ func (s *sensorServer) ListSensors(ctx context.Context, in *sensorpkg.ListSensor
 	return list, nil
 }
 
+func (s *sensorServer) GetSensor(ctx context.Context, in *sensorpkg.GetSensorRequest) (*sv1.Sensor, error) {
+	client := auth.GetSensorClient(ctx)
```
Comment (alexec) [inline server/sensor/sensor_server.go:31]: add tests to ArgoServerSuite?

Comment (whynowy) [reply server/sensor/sensor_server.go:31]: done.

File: ui/src/app/sensors/components/sensor-logs-viewer.tsx (hunk 5 above)
Comment (whynowy) [reply ui/src/app/sensors/components/sensor-logs-viewer.tsx:24]: done.

File: server/sensor/sensor_server.go [hunk 7]
```diff
@@ -28,6 +27,29 @@ func (s *sensorServer) ListSensors(ctx context.Context, in *sensorpkg.ListSensor
...
+	return client.ArgoprojV1alpha1().Sensors(in.Namespace).Get(in.Name, metav1.GetOptions{})
```
Comment (simster7) [inline server/sensor/sensor_server.go:32]: Should we also add the upgraded k8s-go library in this PR?

**Review by simster7 — COMMENTED** [review]
LGTM, I'll leave the final review with Alex

**Comment by alexec** [comment]
Let's push things forward!

**Pull request merged**